src/sst/elements/osseous/rtlcomponent.cc

```diff
@@ -0,0 +1,574 @@
+       output.verbose(CALL_INFO, 1, 0, "Memory interface to be loaded is: %s\n", interfaceName.c_str());
+       
+       Params interfaceParams = params.find_prefix_params("memoryinterfaceparams.");
+       interfaceParams.insert("port", "RtlCacheLink");
+       cacheLink = loadAnonymousSubComponent<Interfaces::SimpleMem>(interfaceName, "memory", 0, ComponentInfo::SHARE_PORTS | ComponentInfo::INSERT_STATS,
+               interfaceParams, timeConverter, new SimpleMem::Handler<Rtlmodel>(this, &Rtlmodel::handleMemEvent));
+       
+       if (!cacheLink)
+           output.fatal(CALL_INFO, -1, "%s, Error loading memory interface\n", getName().c_str());
+   }
+
+    std::string memorymanager = params.find<std::string>("memmgr", "rtl.MemoryManagerSimple");
+    if (NULL != (memmgr = loadUserSubComponent<RtlMemoryManager>("memmgr"))) {
+        output.verbose(CALL_INFO, 1, 0, "Loaded memory manager: %s\n", memmgr->getName().c_str());
+    } else {
+        // Warn about memory levels and the selected memory manager if needed
+        if (memorymanager == "rtl.MemoryManagerSimple" /*&& memLevels > 1*/) {
+            output.verbose(CALL_INFO, 1, 0, "Warning - the default 'rtl.MemoryManagerSimple' does not support multiple memory levels. Configuring anyways but memorylevels will be 1.\n");
+            params.insert("memmgr.memorylevels", "1", true);
+        }
+
+        output.verbose(CALL_INFO, 1, 0, "Loading memory manager: %s\n", memorymanager.c_str());
+        Params mmParams = params.find_prefix_params("memmgr.");
+        memmgr = loadAnonymousSubComponent<RtlMemoryManager>(memorymanager, "memmgr", 0, ComponentInfo::SHARE_NONE | ComponentInfo::INSERT_STATS, mmParams);
+        if (NULL == memmgr) output.fatal(CALL_INFO, -1, "Failed to load memory manager: %s\n", memorymanager.c_str());
+    }
+
+    output.verbose(CALL_INFO, 1, 0, "RTL Memory manager construction is completed.\n");
+
+   pendingTransactions = new std::unordered_map<SimpleMem::Request::id_t, SimpleMem::Request*>();
+   pending_transaction_count = 0;
+   unregisterClock(timeConverter, clock_handler);
+   isStalled = true;
+
+    statReadRequests  = registerStatistic<uint64_t>( "read_requests");
+    statWriteRequests = registerStatistic<uint64_t>( "write_requests");
+    statReadRequestSizes = registerStatistic<uint64_t>( "read_request_sizes");
+    statWriteRequestSizes = registerStatistic<uint64_t>( "write_request_sizes");
+    statSplitReadRequests = registerStatistic<uint64_t>( "split_read_requests");
+    statSplitWriteRequests = registerStatistic<uint64_t>( "split_write_requests");
+    statFlushRequests = registerStatistic<uint64_t>( "flush_requests");
+    statFenceRequests = registerStatistic<uint64_t>( "fence_requests");
+
+
+   // Tell SST to wait until we authorize it to exit
+   registerAsPrimaryComponent();
+   primaryComponentDoNotEndSim();
+
+   assert(ArielRtlLink);
```

Comment:
You could replace this with `sst_assert(ArielRtlLink, CALL_INFO, -1, "ArielRtlLink is null");` so that this will give better error output if it fails. See `sst-core/src/sst/core/baseComponent.h`.